.github/workflows/ci.yml

```diff
@@ -0,0 +1,73 @@
+name: Garage CI
+
+env:
+  GARAGE_GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  TRAVIS_COMMIT_RANGE: origin/${{ github.base_ref }}...origin/${{ github.head_ref }}
+  DOCKER_TAG: garage-ci-${{ github.run_id }}
+  OWNER: "rlworkgroup"
+  DOCKER_CACHE_REPO: "garage-ci"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build_docker_container:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Login to GitHub Package Registry
+      run: echo ${{ secrets.CI_REGISTRY_TOKEN }} | docker login docker.pkg.github.com -u gitanshu --password-stdin
+    - name: Build Docker container
+      run: |
+        docker build . \
+          -f docker/Dockerfile \
+          --target garage-dev-18.04 \
+          -t "${DOCKER_TAG}" \
+          --build-arg GARAGE_GH_TOKEN \
+          --cache-from="docker.pkg.github.com/${OWNER}/${DOCKER_CACHE_REPO}/${DOCKER_TAG}"
+    - name: Push to cache (GitHub Package Registry)
+      run: |
+        docker tag "${DOCKER_TAG}" "docker.pkg.github.com/${OWNER}/${DOCKER_CACHE_REPO}/${DOCKER_TAG}"
+        docker push "docker.pkg.github.com/${OWNER}/${DOCKER_CACHE_REPO}/${DOCKER_TAG}"
+
+
+  check_pre_commit:
```

Comment:
can you name these with human-friendly titles, so they show up a little nicer in the Web UI?